docs/guidance/aws-eks-iam.md

```diff
@@ -0,0 +1,68 @@
+# IAM roles for service account on AWS EKS
+
+Applications in a pod's containers can use an AWS SDK or the AWS CLI to make API requests to AWS services using AWS Identity and Access Management (IAM) permissions. Applications must sign their AWS API requests with AWS credentials. IAM roles for service accounts provide the ability to manage credentials for your applications. To achieve this, you can read the following articles:
+
+* [IAM roles for service accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html): This is the official AWS documentation that explains IAM roles for service accounts step-by-step.
+* [Understanding IAM roles for service accounts, IRSA, on AWS EKS.](https://medium.com/@ankit.wal/the-how-of-iam-roles-for-service-accounts-irsa-on-aws-eks-3d76badb8942): Good article to provide an easy-to-understand explanation.
+
+## Pitfall
+
+For example, users may want to download their files from their S3 bucket with AWS Python SDK (`boto3`) in Ray Pods. However, there is a pitfall in the Ray images. When you execute the **boto3_example_1.py** in a Ray Pod, you will get an error like `An error occurred (403) when calling the HeadObject operation: Forbidden`.
+
+```python
+# boto3_example_1.py
+import boto3
+
+s3 = boto3.client('s3')
+bucket_name = YOUR_BUCKET_NAME
+key = YOUR_OBJECT_KEY
+filename = YOUR_FILENAME
+
+s3.download_file(bucket_name, key, filename)
+```
+
+The root cause is that the version of `boto3` in the Ray image is too old. To elaborate, `rayproject/ray:2.3.0` provides boto3 version 1.4.8 (Nov. 21, 2017),
+a more recent version (1.26) is currently available as per https://pypi.org/project/boto3/#history. The boto3 1.4.8 does not support to initialize the security credentials automatically in some cases (e.g. `AssumeRoleWithWebIdentity`). 
+
+```shell
+# image: rayproject/ray:2.3.0
+pip freeze | grep boto
+# boto3==1.4.8
+# botocore==1.8.50
+```
+
+## Workaround solutions
```

Comment:
Also a use-case where things will break - if you use the ray image  < `ray:2.3.0`  in a `RayService` and set the working directory to a zip file in a private S3 bucket, the Ray Serve application will not be able to start.

I have not tested this yet, but a potential solution could be to 
1. Create a custom docker image on top of `ray:2.3.0` and manually install the latest `boto3` version (would defer to the Ray team to make sure this works)
2. Create a custom docker image on top of `ray:2.3.0`, add your code in the image, and set the working directory (in `serveConfig`) to a local directory. This should get serve application running. Any model deployment code trying to access S3 should do that using **boto3_example_2.py**. (I will be testing this in the next few days, but will defer to the Serve team to make sure this works).

Reply:
<img width="1416" alt="Screen Shot 2023-03-13 at 3 09 38 PM" src="https://user-images.githubusercontent.com/20109646/224843238-82de6e69-a394-43fd-957d-3743c6dfcd98.png">

I tried to update `working_dir` to point to my private S3 bucket, and got the error above. We need to update the boto3 version.

Reply:
Got it! Thanks for letting me know!

Reply:
@architkulkarni is working on updating the boto3 version: https://github.com/ray-project/ray/pull/33273.